### PR TITLE
Adds gender TF chemicals + Spontaneous TF pref

### PR DESCRIPTION
**Pull request by Heroman3003**
In theory Spontaneous TF pref and the gender tf chems could be expanded later but this is the easiest and probably simultaneously most 'universally desired' kind of tf mechanic, so just have that.

Adds a new 'base' chem (similar to sizeoxadone) - amorphorovir. Can be mixed with other reagents to make actual drugs themselves.
For now only three drugs are available. This is more for-discussion/proof of concept thing in case so it's really simple. Androrovir for Male tgtf, Gynorovir for Female tgtf and Androgynorovir for Intersex (also known by byond as PLURAL) tgtf. All ONLY change the gender and gender identity. If species has no sexual dymorphism there will be no visual change at all, and even if it does, other than shape of their (naked) chest nothing will change (gotta use stuff like mirrors for that).

There are two recipies for each chem. All involve amorphorovir as base, but one recipe is for 'chemistry' production, and another for 'bootleg', using reagents obtainable with public access and a bit of hassle. A vial with 1 unit of amorphorovir is added as rare-ish maint loot spawn.

List of actual recipies:
~~Amorphorovir - 30 parts clonexadone, 30 parts biomass, 5 parts sizeoxadone, 5 parts phoron = 1 part amorphorovir~~
Amorphorovir - 30 parts cryptobiolin, 30 parts biomass, 20 parts hyperzine, 5 parts phoron (catalyst) = 1 part amorphorovir
Androrovir - 1 part amorphorovir, 20 parts bicaridine, 20 parts iron, 20 parts ethanol = 1 part androrovir
Gynorovir - 1 part amorphorovir, 20 parts inaprovaline, 20 parts silicon, 20 parts sugar = 1 part gynorovir
Androgynorovir - 1 part amorphorovir, 20 parts dylovene, 20 parts fluorine, 20 parts tungsten= 1 part androgynorovir
Androrovir (homemade) - 1 part amorphorovir, 10 units animal protein, 10 units capsaicin = 1 part androrovir
Gynorovir (homemade) - 1 part amorphorovir, 10 units soymilk, 10 units sugar = 1 part gynorovir
Androgynorovir (homemade) - 1 part amorphorovir, 10 units space cola, 10 units berry juice = 1 part androgynorovir

Spontaneous TF pref can be set in Vore Panel and is OFF by default. It prevents any of the above chems, as well as advanced mutation toxin (one that tfs you into prommie) from having any effect on you at all.

### CODE REVIEW

**Comment by Huenererschrecker** [comment]
Finaly, a way to make the impossible whopper

**Comment by Screemonster** [comment]
There _has_ to be a botany plant for growing this stuff, if only for the meme of someone drinking a soy latte and immediately growing tits.

**Comment by Heroman3003** [comment]
> 
> 
> There _has_ to be a botany plant for growing this stuff, if only for the meme of someone drinking a soy latte and immediately growing tits.

~~the reference already exists in the 'makeshift' version of the Gynorovir recipe~~

**Comment by ArgobargSoup** [comment]
I would remove the part that sets gender identity, and rephrase the TF message ta mention them looking physically <new gender>.  That bein' said, I do quite like this~

...Also bonuses if it actually updates the character sprite.

**Comment by lm40** [comment]
i don't support this on my general principle of not supporting the enshrinement of non-vore-related fetish mechanics

**Comment by Wickedtemp** [comment]
Having non-vore-specific mechanics is good for the server. We already have size-changing guns and chems because not **everything** has to be tied to vore. Sometime's I'm not in the _mood_ for vore, but I _am_ in the mood for TF. Taur riding isn't a vore mechanic, but it's fun as hell and is totally used for fetish reasons sometimes. We have maid outfits, latex, and I'm pretty sure the riding crop's still in the game as well, and all of that adds different things to the server that others would enjoy the presence of. If we only ever did vore and never did anything that wasn't vore, it'd be...pretty boring. No reason to restrict things so harshly. 

Opinion aside, the reagents recipes look fine. Fun references and this provides an IC means of changing things that some players may be interested in changing in-round.

**Comment by Arokha** [comment]
> size-changing guns and chems

Were added because of vore

> Taur riding isn't a vore mechanic [...] and I'm pretty sure the riding crop's still in the game as well

I added these as a joke. You suggested the riding crop because taur riding uses vehicle mechanics and holding a key let you drive the vehicle, so as the key for taurs we decided on riding crop because it was funny. They might be someone's fetish but I coded them because I was feeling silly and as a joke.

> maid outfits

/tg/ has a xenomorph in a maid outfit, it's not fetish content. The latex though is something I personally wish wasn't added for the same reason as lm40 says, though.

**Comment by Very-Soft** [comment]
Not my cup of tea but, it comes with a preference toggle defaulted to off. This is an opt in thing, so, imo, fine as long as it stays that way.

**Comment by pontifexminimus2** [comment]
well obviously I am going to be biased here.

Uhhh... Yes? 😅

**Comment by ImABigFag** [comment]
> well obviously I am going to be biased here.
> 
> Uhhh... Yes

> well obviously I am going to be biased here.
> 
> Uhhh... Yes? 😅

Is this even a question for you?

**Comment by ImABigFag** [comment]
Anyway. I've put in my 100% support for this.

File: code/modules/reagents/reagents/vore_vr.dm [hunk 1]
```diff
@@ -117,16 +117,74 @@
...
+		return
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(M.reagents.has_reagent("gynorovir") || M.reagents.has_reagent("androgynorovir"))
+			H.Confuse(1)
+		else
+			if(!(H.gender == MALE))
+				H.set_gender(MALE)
+				H.change_gender_identity(MALE)
+				H.visible_message("<span class='notice'>[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now male.</span>",
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:147]:
```suggestion
				H.visible_message(SPAN_NOTICE("[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now male."),
```

File: code/modules/reagents/reagents/vore_vr.dm [hunk 2]
```diff
@@ -117,16 +117,74 @@
...
+								"<span class='warning'>Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you just became male.</span>")
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:148]:
```suggestion
								SPAN_WARNING("Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you have become male."))
```

File: code/modules/reagents/reagents/vore_vr.dm [hunk 3]
```diff
@@ -117,16 +117,74 @@
...
+		return
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(M.reagents.has_reagent("androrovir") || M.reagents.has_reagent("androgynorovir"))
+			H.Confuse(1)
+		else
+			if(!(H.gender == FEMALE))
+				H.set_gender(FEMALE)
+				H.change_gender_identity(FEMALE)
+				H.visible_message("<span class='notice'>[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now female.</span>",
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:168]:
```suggestion
				H.visible_message(SPAN_NOTICE("[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now female."),
```

File: code/modules/reagents/reagents/vore_vr.dm [hunk 4]
```diff
@@ -117,16 +117,74 @@
...
+								"<span class='warning'>Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you just became female.</span>")
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:169]:
```suggestion
								SPAN_WARNING("Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you have become female."))
```

File: code/modules/reagents/reagents/vore_vr.dm [hunk 5]
```diff
@@ -117,16 +117,74 @@
...
+		return
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(M.reagents.has_reagent("gynorovir") || M.reagents.has_reagent("androrovir"))
+			H.Confuse(1)
+		else
+			if(!(H.gender == PLURAL))
+				H.set_gender(PLURAL)
+				H.change_gender_identity(PLURAL)
+				H.visible_message("<span class='notice'>[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now of mixed gender.</span>",
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:189]:
```suggestion
				H.visible_message(SPAN_NOTICE("[H] suddenly twitches as some of their features seem to contort and reshape, adjusting... In the end, it seems they are now androgynous."),
```

File: code/modules/reagents/reagents/vore_vr.dm [hunk 6]
```diff
@@ -117,16 +117,74 @@
...
+								"<span class='warning'>Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you just became of mixed gender.</span>")
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagents/vore_vr.dm:190]:
```suggestion
								SPAN_WARNING("Your body suddenly contorts, feeling very different in various ways... By the time the rushing feeling is over it seems you have become androgynous."))
```

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
Unsure if it's in scope, but it may be a good idea to add a pref for whether to change just the character's `gender` var, or to change both the `gender` var and their pronouns, as suggested by someone else on this PR.
Otherwise, only concerns are style and some rewording.

**Comment by Screemonster** [comment]
personally my position on non-vore kinks is that as long as they're not obtrusive (like messes that some other poor fucker has to clean up) I don't give a rat's ass.
the original version of the latex clothing stuff was eh for me 'cause I was worried that someone would go wearing it in a context where people are forced to interact with it, same with that PR for visible genitalia sprites that someone tried to add a while back (which I remember commenting on the PR that I wasn't keen on merging any change that contained the phrase "vagina, dripping") because with those there's a real chance that someone's gonna just have them out while they're doing their job on the station and some poor sod is going to rock up to R&D to ask for a diamond drill or something and be forced to behold the kebab
but sexchange chemicals? Don't we already have sexchange-tf as a vore thing and it's pretty much exclusively done in scenes where people don't have to look at it? It doesn't strike me as the kind of thing people are going to be forced to be involved in.

**Review by Novacat — CHANGES_REQUESTED** [review]
I would rather the phoron be used as a catalyst rather than an ingredient. 5 Phoron is a lot and I can already see happening with this the same thing that occasionally happens with microcillin/macrocillin where a meme chemist blows all of medical's phoron on TF chems.

**Comment by MarinaGryphon** [comment]
> I would rather the phoron be used as a catalyst rather than an ingredient. 5 Phoron is a lot and I can already see happening with this the same thing that occasionally happens with microcillin/macrocillin where a meme chemist blows all of medical's phoron on TF chems.

Yeah, that's one concern I didn't address in my review—it seems rather expensive. You need to use 30 clonexadone and 30 biomass to get _one_ unit of the precursor. That's just not sustainable if you have more than one person asking for it.

Honestly, I think the suggestion of a botany plant for it would be great.

File: code/modules/reagents/reagents/vore_vr.dm (hunk 2 above)
Comment (ShadowLarkens) [reply code/modules/reagents/reagents/vore_vr.dm:148]: Suggestion: Dismiss this review.


The macros are less efficient and don't provide much benefit (why would you ever want to change the class name of warning?)

**Comment by foopwo** [comment]
> 
> 
> > I would rather the phoron be used as a catalyst rather than an ingredient. 5 Phoron is a lot and I can already see happening with this the same thing that occasionally happens with microcillin/macrocillin where a meme chemist blows all of medical's phoron on TF chems.
> 
> Yeah, that's one concern I didn't address in my review—it seems rather expensive. You need to use 30 clonexadone and 30 biomass to get _one_ unit of the precursor. That's just not sustainable if you have more than one person asking for it.
> 
> Honestly, I think the suggestion of a botany plant for it would be great.

The size chems are in the same boat. The recipe is awkward and making more than a couple doses is costly.
Meanwhile size guns are literally found in the trash. I've literally never seen the chems used in favor of a gun.
Both the size chems and these really should have their recipes revised.

**Comment by lm40** [comment]
i think slippery slope argument is valid wrt to fetish mechanics because every time a new one is proposed people trot out 'x fetish mechanic already exists in the game so why not add z fetish mechanic', and there is then no non-arbitrary reason to deny it. i would rather not see fetish mechanics that have literally nothing to do with vore added for this reason.

i view it as much the  same way as if you wanted to like, roleplay sex or whatever. you are free to roleplay it if you want but it doesn't need to have associated game mechanics. this will probably be added tho so i really see no reason to belabor the point, i'm just clarifying the reason for my perspective/opposition.

**Comment by Wickedtemp** [comment]
The recipes are fine. Phoron is generally plentiful. Currently, Medical starts with 200u due to both Asteroid and Surface still existing together.

Most reagent stocks don't use more than 80u. I personally don't use more than 60u. 

There's also spare phoron to be found in Science and Engineering that Medical can ask for, even without Cargo. I doubt this would be a Hot Item thats requested five times every shift, so the phoron cost, in my opinion, is rather balanced. 

As for using biomass and sizeoxadone, well...biomass is incredibly cheap and easy to create, and while sizeoxadone appears to be complicated, there are methods to creating it quickly that I'll post here in a lil bit. 

The primary reasons the sizeguns are preferred over size changing reagents are because it's an instant size change and you can set the value exactly. It's just easier and more convenient and you can get larger or smaller with one device, while chemically you'd need two or three different chems to get the same utility. Phoron cost or recipe doesn't really factor in. 

Also the slippery-slope is a fallacy for a reason. "Where does it stop?!" - somewhere. We aren't even adding anything new. These mechanics already exist. It's just not tied to vore, and there's nothing inherently wrong about that.

**Comment by H0lySquirr3l** [comment]
> i think slippery slope argument is valid wrt to fetish mechanics because every time a new one is proposed people trot out 'x fetish mechanic already exists in the game so why not add z fetish mechanic', and there is then no non-arbitrary reason to deny it. i would rather not see fetish mechanics that have literally nothing to do with vore added for this reason.
> 
> i view it as much the same way as if you wanted to like, roleplay sex or whatever. you are free to roleplay it if you want but it doesn't need to have associated game mechanics. this will probably be added tho so i really see no reason to belabor the point, i'm just clarifying the reason for my perspective/opposition.



>>>
This is fair, if this was a whole new mechanic I would agree that It would require a bit more discussion on if it would really need a mechanical connection outside of just RP.

File: code/modules/reagents/reagents/vore_vr.dm (hunk 1 above)
Comment (Heroman3003) [reply code/modules/reagents/reagents/vore_vr.dm:147]: These macros aren't used anywhere else and unless the code as a whole is getting converted to them, I'd prefer using the format that's more universally standard as of now.

**Comment by Heroman3003** [comment]
> 
> 
> I would remove the part that sets gender identity, and rephrase the TF message ta mention them looking physically . That bein' said, I do quite like this~
> 
> ...Also bonuses if it actually updates the character sprite.



> 
> 
> Unsure if it's in scope, but it may be a good idea to add a pref for whether to change just the character's `gender` var, or to change both the `gender` var and their pronouns, as suggested by someone else on this PR.
> Otherwise, only concerns are style and some rewording.

Main reason for change of pronoun is simple. If you don't like it you can immedeately fix it yourself after the fact, as you can actually change 'gender identity' in-game with a verb... But otherwise? In a lot of cases, the 'gender change' would have pretty much no visual effects at all, due to sprites being either extremely similar or just having no dymorphism. So, in terms of visual effects, without gender identity part, if you do a gender change, to anyone it'd be no different from just having a single message printed in chat, as both in Examines and  visually (unless naked and of dymorphic species) they'd be the same.

**Comment by Heroman3003** [comment]
> 
> 
> I would rather the phoron be used as a catalyst rather than an ingredient. 5 Phoron is a lot and I can already see happening with this the same thing that occasionally happens with microcillin/macrocillin where a meme chemist blows all of medical's phoron on TF chems.
> 
> PS: The use of Biomass is pretty icky as well.

I think the recipe is fine, although I do agree that maybe replacing sizeoxadone OR phoron could be sensible. Same as size chems it's meant to be something that is easy to make with starting chem, in small amounts, but impossible to mass-produce. 1 unit = 1 TF already, and I doubt any scene would need more than one. You can also find 1-unit vials in maint. As for "but size chems aren't made", I do agree that its mostly because of just how much worse they are unless you are specifically going for exactly the chem-based scene, any 'generic' size scene is better off with sizeguns purely for convinience.

**Comment by Iroquois-Pliskin** [comment]
lmao all of a sudden there are people worried about adding fetish content after we just added boozehounds

Sure why not. It's tame by comparison

**Comment by dwinters99** [comment]
this already exists as a belly option so I don't see why it shouldn't be a chem
slippery slope while a reasonable point is maybe misplaces here because as stated it already exists in the form of belly tf menu

**Comment by Novacat** [comment]
It should be easier on the Phoron and phoron-based chems but otherwise is fine.

**Comment by Heroman3003** [comment]
> 
> 
> It should be easier on the Phoron and phoron-based chems but otherwise is fine.

Done. Still uses phoron chems, but way less. New recipe is:

30 cryptobiolin + 30 biomass + 15 hyperzine + 5 phoron = 1 amorphorovir

**Review by Novacat — APPROVED** [review]

**Pull request merged**